test/unit_test/agent/test_dsl_topology_consistency.py

```
@@ -0,0 +1,195 @@
+
+
+def test_executable_graph_node_without_component_is_rejected():
+    dsl = {
+        "components": {"begin": _component("Begin")},
+        "graph": {
+            "nodes": [_node("begin", "Begin"), _node("quality", "Quality")],
+            "edges": [{"source": "begin", "target": "quality"}],
+        },
+    }
+
+    with pytest.raises(ValueError, match="missing components=.*quality"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)